node/resolvers/catalog/index.ts

```diff
@@ -184,4 +184,17 @@ export const queries = {
     })
     return map(resolveCategoryFields, categories)
   },
+
+  search: async (_, data, { vtex: ioContext }: ColossusContext, info) => {
+    const { map, query, rest } = data
+    const facetsMap = map.split(',').slice(0, query.split('/').length).join(',')
+    const queryWithRest = query + (rest && '/' + rest.replace(/,/g, '/'))
+    const facetsValue = query + '?map=' + facetsMap
+    const facetsValueWithRest = queryWithRest + '?map=' + map
+    const products = await queries.products(_, { ...data, query: queryWithRest }, { vtex: ioContext }, info)
```

Comment:
I think lines 194-196 can be run in parallel. If that's the case, why not making root resolvers or Promise.all in an array ? It would make the code run faster